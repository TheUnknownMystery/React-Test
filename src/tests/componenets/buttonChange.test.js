import renderer, { act } from 'react-test-renderer';
import ButtonChange from '../../components/buttonChange';

test('render change button', () => {
  const buttonChange = renderer.create(<ButtonChange text="Change" />);
  act(() => {
    // eslint-disable-next-line testing-library/await-async-query
    buttonChange.root.findByType('button').props.onClick();
  });

  // eslint-disable-next-line testing-library/await-async-query
  expect(buttonChange.root.findByType('button').children[0]).toBe('Change');
  expect(buttonChange.toJSON()).toMatchSnapshot();
});

test('render button with default props', () => {
  const buttonChange = renderer.create(<ButtonChange />);
  act(() => {
    // eslint-disable-next-line testing-library/await-async-query
    buttonChange.root.findByType('button').props.onClick();
  });
  // eslint-disable-next-line testing-library/await-async-query
  expect(buttonChange.root.findByType('button').children[0]).toBe(
    'Changed Text'
  );
  expect(buttonChange.toJSON()).toMatchSnapshot();
});
