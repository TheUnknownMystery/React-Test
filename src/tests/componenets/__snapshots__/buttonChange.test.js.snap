// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button with default props 1`] = `
<button
  onClick={[Function]}
>
  Changed Text
</button>
`;

exports[`render change button 1`] = `
<button
  onClick={[Function]}
>
  Change
</button>
`;
